ni_measurementlink_service/session_management.py

```diff
@@ -418,6 +444,520 @@ def create_sessions(
+                session. If this argument is not specified, the default value is
+                an empty dict, which you may override by specifying
+                ``NIFGEN_SIMULATE``, ``NIFGEN_BOARD_TYPE``, and ``NIFGEN_MODEL``
+                in the configuration file (``.env``).
+
+            initialization_behavior: Specifies whether to initialize a new
+                session or attach to an existing session.
+
+        Returns:
+            A context manager that yields a session information object. The
+            created session is available via the ``session`` field.
+
+        See Also:
+            For more details, see :py:class:`nifgen.Session`.
+        """
+        from ni_measurementlink_service._drivers._nifgen import SessionConstructor
+
+        session_constructor = SessionConstructor(
+            self._discovery_client,
+            self._grpc_channel_pool,
+            reset_device,
+            options,
+            initialization_behavior,
+        )
+        return self._create_session_core(session_constructor, INSTRUMENT_TYPE_NI_FGEN)
+
+    @requires_feature(SESSION_MANAGEMENT_2024Q1)
+    def create_nifgen_sessions(
+        self,
+        reset_device: bool = False,
+        options: Optional[Dict[str, Any]] = None,
+        initialization_behavior: SessionInitializationBehavior = SessionInitializationBehavior.AUTO,
+    ) -> ContextManager[Sequence[TypedSessionInformation[nifgen.Session]]]:
+        """Create multiple NI-FGEN instrument sessions.
+
+        Args:
+            reset_device: Specifies whether to reset the instrument during the
+                initialization procedure.
+
+            options: Specifies the initial value of certain properties for the
+                session. If this argument is not specified, the default value is
+                an empty dict, which you may override by specifying
+                ``NIFGEN_SIMULATE``, ``NIFGEN_BOARD_TYPE``, and ``NIFGEN_MODEL``
+                in the configuration file (``.env``).
+
+            initialization_behavior: Specifies whether to initialize a new
+                session or attach to an existing session.
+
+        Returns:
+            A context manager that yields a sequence of session information
+            objects. The created sessions are available via the ``session``
+            field.
+
+        See Also:
+            For more details, see :py:class:`nifgen.Session`.
+        """
+        from ni_measurementlink_service._drivers._nifgen import SessionConstructor
+
+        session_constructor = SessionConstructor(
+            self._discovery_client,
+            self._grpc_channel_pool,
+            reset_device,
+            options,
+            initialization_behavior,
+        )
+        return self._create_sessions_core(session_constructor, INSTRUMENT_TYPE_NI_FGEN)
+
+    @requires_feature(SESSION_MANAGEMENT_2024Q1)
+    def create_niscope_session(
+        self,
+        id_query: bool = False,
```

Comment:
id_query isn't an option for the gRPC Initialize in LabVIEW. I think this is true for all MI drivers with gRPC support. I don't know why that support wasn't added, but we should make sure that this works.

Reply:
id_query isn't very important for PXI modular instruments. Perhaps we should remove it.

Reply:
I checked with @marcoskirsch and he agrees with removing `id_query`.